esp-wifi/src/wifi/mod.rs

```diff
@@ -245,12 +248,11 @@ pub struct AccessPointInfo {
 }
 
 /// Configuration for a Wi-Fi access point.
-#[derive(Clone, Debug, PartialEq, Eq)]
-#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+#[derive(Clone, PartialEq, Eq)]
 #[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
 pub struct AccessPointConfiguration {
     /// The SSID of the access point.
-    pub ssid: heapless::String<32>,
+    pub ssid: alloc::string::String,
```

Comment:
I can't find where this is being used to verify but would there be any benefits for this to be `CString` instead of `String`?

Reply:
not too sure how a CString in a public API feels like

Reply:
Ah I found it. https://github.com/esp-rs/esp-hal/blob/dca82d60b1dbeb18fb9035790737cd8f9fa8fc50/esp-wifi/src/wifi/mod.rs#L2114. Looks like CString isn't necessary as it's copied.

Though I feel like there should be some explicit bounds checking on that now.

Reply:
We probably want to return an error instead of causing an OOB-panic, yes